flake.nix

```diff
@@ -0,0 +1,88 @@
+{
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs = inputs@{ nixpkgs, flake-parts, rust-overlay, ... }:
+    flake-parts.lib.mkFlake {
+      inherit inputs;
+    } {
+      systems = [
+        "x86_64-linux"
+	"i686-linux"
+	"aarch64-linux"
+      ];
+      perSystem = {config, system, ...}: let
+        pkgs = import nixpkgs { 
+	  inherit system; 
+	  overlays = [ (import rust-overlay) ];
+	};
+	rust-toolchain = "stable";
+	rust-package = pkgs.rust-bin."${rust-toolchain}".latest.default;
+	
+	native-dependencies = with pkgs; [
+           pkg-config
+	   makeWrapper
+	];
+	
+	dependencies = with pkgs; [
+	    libglvnd
+	    libseat
+	    wayland
+	    wayland-scanner
+	    pkg-config
+	    libdrm
+	    systemdLibs # Contains libudev. DON'T PANIC: it won't install the whole init system
+	    libxkbcommon
+	    mesa
+            libinput
+	    xorg.libX11 # Needed for xwayland to work
+	    xorg.libXcursor
+	    xorg.libXi
+	    
+	  ];
+
+          # TODO: code this in a split file and import it
+	  magmawm = pkgs.rustPlatform.buildRustPackage {
+	    rust = rust-package;
+            pname = "magmawm";
+	    version = "0.0.1";
+	    src = ./.;
+	    buildInputs = dependencies;
+	    nativeBuildInputs = native-dependencies;
+
+	    cargoLock = {
+              lockFile = ./Cargo.lock;
+	      outputHashes = {
+                "smithay-0.3.0" = "sha256-UQwYV3GOkZHEFyAMRMY46JiFXsdsjfSVHAHlA47ZAtI=";
```

Comment:
We are not using the release version of smithay, we use the git version.

Reply:
I am aware. The version number i put was the crate version number, not the github release version number.

Reply:
we use the git crate though, im unsure what you mean here

```toml
[dependencies.smithay]
git = "https://github.com/Smithay/smithay.git"
default-features = false
features = [
    "backend_winit",
    "wayland_frontend",
    "desktop",
    "backend_session_libseat",
    "backend_drm",
    "renderer_multi",
    "renderer_glow",
    "backend_gbm",
    "backend_udev",
    "backend_egl",
    "backend_libinput",
    "use_system_lib",
]

```

Reply:
We do use the git crate, but crates have a version field in their Cargo.toml field. On smithay's github, the version field for the cargo project is 0.3.0